From d653ede5127c571b751033d68805e6a899f58027 Mon Sep 17 00:00:00 2001
From: Pierre-Jean Texier <pjtexier@koncepto.io>
Date: Sat, 16 Mar 2019 13:35:22 +0000
Subject: [PATCH] swuforward: fix install_remote_swu() on the first update

Check if remote update was successful only if 'ignore' is false.
This to avoid the following message on the first update :

[ERROR] : SWUPDATE failed [0] ERROR handlers/swuforward_handler.c : retrieve_msgs : 281 : Update to http://10.0.0.100:8080 was NOT successful !

Signed-off-by: Pierre-Jean Texier <pjtexier@koncepto.io>
---
 handlers/swuforward_handler.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/handlers/swuforward_handler.c b/handlers/swuforward_handler.c
index c08edcd..33ccb00 100644
--- a/handlers/swuforward_handler.c
+++ b/handlers/swuforward_handler.c
@@ -277,9 +277,11 @@ static int retrieve_msgs(struct hnd_priv *priv, bool ignore)
 				ret = -ETIMEDOUT;
 			}
 		} while (ret > 0);
-		if (ret != 0 || (conn->SWUpdateStatus != SUCCESS)) {
-			ERROR("Update to %s was NOT successful !", conn->url);
-			result = -1;
+		if (!ignore) {
+			if (ret != 0 || (conn->SWUpdateStatus != SUCCESS)) {
+				ERROR("Update to %s was NOT successful !", conn->url);
+				result = -1;
+			}
 		}
 	}
 
-- 
2.7.4

